sound/core/compress_offload.c

```diff
@@ -488,6 +488,48 @@ snd_compr_get_codec_caps(struct snd_compr_stream *stream, unsigned long arg)
 }
 #endif /* !COMPR_CODEC_CAPS_OVERFLOW */
 
+int snd_compr_malloc_pages(struct snd_compr_stream *stream, size_t size)
+{
+	struct snd_dma_buffer *dmab;
+	int ret;
+
+	if (snd_BUG_ON(!(stream) || !(stream)->runtime))
+		return -EINVAL;
+	dmab = kzalloc(sizeof(*dmab), GFP_KERNEL);
+	if (!dmab)
+		return -ENOMEM;
+	dmab->dev = stream->dma_buffer.dev;
+	ret = snd_dma_alloc_pages(dmab->dev.type, dmab->dev.dev, size, dmab);
+	if (ret < 0) {
+		kfree(dmab);
+		return ret;
+	}
+
+	snd_compr_set_runtime_buffer(stream, dmab);
+	stream->runtime->dma_bytes = size;
+	return 1;
```

Comment:
any particular reason to return 1?

Reply:
As with snd_pcm_lib_malloc_pages, we return 1 if area was changed.

Reply:
Hmm.. now when I think about, I simplified _compr_ equivalent as there are no other use-cases then probes and as stated, in future I'd like to see compr making use of pcm variant. In this simplified version, one could say: "area has always changed". Returning 1 still looks good to me though - sticks with _pcm_lib plan.